ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -748,15 +748,29 @@ private[kvutils] class TransactionCommitter(
 }
 
 private[kvutils] object TransactionCommitter {
-  case class DamlTransactionEntrySummary(submission: DamlTransactionEntry) {
+  class DamlTransactionEntrySummary(val submission: DamlTransactionEntry, tx: => Tx.Transaction) {
     val ledgerEffectiveTime: Timestamp = parseTimestamp(submission.getLedgerEffectiveTime)
     val submitterInfo: DamlSubmitterInfo = submission.getSubmitterInfo
     val commandId: String = submitterInfo.getCommandId
     val submitters: List[Party] =
       submitterInfo.getSubmittersList.asScala.toList.map(Party.assertFromString)
-    lazy val transaction: Tx.Transaction = Conversions.decodeTransaction(submission.getTransaction)
+    lazy val transaction: Tx.Transaction = tx
     val submissionTime: Timestamp = Conversions.parseTimestamp(submission.getSubmissionTime)
     val submissionSeed: crypto.Hash = Conversions.parseHash(submission.getSubmissionSeed)
+    // On copy, avoid decoding the transaction again if not needed
+    def copyPreservingDecodedTransaction(
+        submission: DamlTransactionEntry
+    ): DamlTransactionEntrySummary = {
+      new DamlTransactionEntrySummary(submission, transaction)
```

Comment:
At this point, you are forcing `transaction` regardless, so why not just do it up front and make `tx` a strict (non-lazy) value?

(And rename it to `transaction`, as you will no longer need the `lazy val` later.)

Reply:
I've kept the lazy semantics for historical reasons, since I believe it's an optimization (e.g. `authorizeSubmitters` does not use it so if this step emits `StepStop` it won't ever be needed).

Reply:
On a closer look, `authorizeSubmitters` would cause an abort just in exceptional cases, so I guess there won't be a significant impact if we decode it eagerly.

Reply:
Done

Reply:
> At this point, you are forcing `transaction` regardless, so why not just do it up front and make `tx` a strict (non-lazy) value?
> 
> (And rename it to `transaction`, as you will no longer need the `lazy val` later.)

@SamirTalwar-DA , I was just about to merge this but I double-checked the need for `lazy`. I missed the fact that the `DamlTransactionEntrySummary` is created again in `trimUnnecessaryNodes`, but the decoded transaction is however not needed in the subsequent step `buildFinalLogEntry`. I'll revert to using the lazy implementation